File: patterns/state-machine-driven-groups/rules-python/state_machine_driven_groups_simple.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[ItemStateChangeTrigger("vTimeOfDay")])
class SetLightsBasedOnTimeOfDay:
    def execute(self, module, input):
        time_of_day = str(Registry.getItem("vTimeOfDay").state)
        Registry.getItem("gLights_OFF_" + time_of_day).sendCommand("OFF")
        Registry.getItem("gLights_ON_" + time_of_day).sendCommand("ON")

File: patterns/state-machine-driven-groups/rules-js/state-machine-driven-groups-simple.js
rules.JSRule({
  name: "Set lights based on Time of Day",
  triggers: [triggers.ItemStateChangeTrigger("vTimeOfDay")],
  execute: (event) => {
    const timeOfDay = items.getItem("vTimeOfDay").state;
    items.getItem("gLights_OFF_" + timeOfDay).sendCommand("OFF");
    items.getItem("gLights_ON_" + timeOfDay).sendCommand("ON");
    console.log("vTimeOfDay=" + timeOfDay + " -> Lichtgruppen angesteuert");
  }
});
